sourcecode/apis/contentauthor/app/Libraries/H5P/H5PLibraryAdmin.php

```diff
@@ -44,11 +44,18 @@ public function handleUpload(string $path, bool $upgradeOnly, bool $disableFileC
         // Move so core can validate the file extension.
         rename($path, $newPath);
 
-        if (!$this->validator->isValidPackage(true, $upgradeOnly)) {
-            @unlink($this->framework->getUploadedH5pPath());
+        try {
+            if (!$this->validator->isValidPackage(true, $upgradeOnly)) {
+                @unlink($this->framework->getUploadedH5pPath());
 
+                throw new InvalidH5pPackageException(
+                    $this->validator->h5pF->getMessages('error'),
+                );
+            }
+        } catch (\Exception $e) {
+            @unlink($this->framework->getUploadedH5pPath());
             throw new InvalidH5pPackageException(
-                $this->validator->h5pF->getMessages('error'),
+                [$e->getMessage()],
```

Comment:
I don't agree with turning every \Exception into InvalidH5pPackageException. This exception class exists to convey error messages about uploads back to the user, and as `catch (\Exception)` matches all kinds of errors, we risk leaking sensitive information to the user, or confusing them with irrelevant information.

From what I can tell, `isValidPackage()` shouldn't throw in the first place, so when this happens it is due to programming errors. If we're going to handle these anyway, it'd be better to first log the causing exception, then throw InvalidH5pPackageException with a generic error message.